src/rdkafka_broker.c

```diff
@@ -3605,6 +3639,8 @@ rd_kafka_fetch_reply_handle (rd_kafka_broker_t *rkb,
 			struct rd_kafka_toppar_ver *tver, tver_skel;
                         rd_kafka_toppar_t *rktp;
                         shptr_rd_kafka_toppar_t *s_rktp = NULL;
+                        rd_avl_t *aborted_txn_offsets = NULL;
+                        rd_list_t aborted_txn_list;
```

Comment:
keeping the list around is only used as an optimization in the destroy method, which is relatively local to here. it's not passed into rd_kafka_msgset_parse  (it shouldn't be either). i'm not sure what functionality you want comparmentalized and unit tested. you're right, it's slightly messy, but I don't see a big enough advantage to encapsulating it up. my original implementation here avoided this complexity but was less efficient because it also created the list in the destroy method as well. that's actually my (slight) preference because it makes the code simpler to read, and the perf. downside is hit only infrequently and wouldn't be of practical consequence, but either or...

Reply:
yes, great idea, this cleans things up a lot.